adr/ADR-15.md

```diff
@@ -177,6 +177,20 @@ Why not append those new tokens at the end? This is to simplify the export/impor
 
 The account hash is not used by the client at this time, only used for routing, same for the last (12th) token.
 
+### Ack Handling
+
+An ack is accomplished by publishing a message using the JSAck as a subject 
+with the payload consisting of the bytes of the Ack Type
+
+Clients must support 3 terminal Ack Types, `+ACK`, `-NAK` and `+TERM` and the progress Ack Type `+WPI`.
+
+Once a terminal Ack Type has been published for a message, the client should ignore all other 
+user requests to send any type of ack. 
```

Comment:
```suggestion
Clients should mark that the client has already acked a message, with the exception of the +WPI.  This will help specially with the issue of auto acking and async callbacks, in case NAK, TERM, or ACK is called in the callback, then the client will notice the mark that the client has already attempted to ack and skip sending another ack.  Acking twice using the same message struct may throw an `nats: message was already acknowledged` error, which in case of async callbacks that is doing auto acking would be discarded.
```

Reference to Go implementation: https://github.com/nats-io/nats.go/blob/main/js.go#L2589-L2593

Reply:
Do we have to put it in the adr that the client has to track this, I think it's clear that clients will have to track it somehow and I know the C/Java/.NET already do exactly this.
I think we have also agreed for the client to silently ignore repeated calls and not return an error